spec/requests/v1/submissions_spec.rb

```diff
@@ -147,34 +147,68 @@
         )
       end
 
-      it 'marks the submission as complete' do
-        post "/v1/submissions/#{submission.id}/complete", headers: { 'X-Auth-Id' => user.auth_id }
+      context 'with no previous completed submission against the task' do
+        it 'marks the submission as complete' do
+          post "/v1/submissions/#{submission.id}/complete", headers: { 'X-Auth-Id' => user.auth_id }
 
-        expect(response).to be_successful
+          expect(response).to be_successful
 
-        submission.reload
+          submission.reload
 
-        expect(submission).to be_completed
-        expect(submission.task).to be_completed
-      end
+          expect(submission).to be_completed
+          expect(submission.task).to be_completed
+        end
 
-      it 'records the user who completed the submission' do
-        post "/v1/submissions/#{submission.id}/complete", headers: { 'X-Auth-Id' => user.auth_id }
+        it 'records the user who completed the submission' do
+          post "/v1/submissions/#{submission.id}/complete", headers: { 'X-Auth-Id' => user.auth_id }
 
-        submission.reload
+          submission.reload
+
+          expect(submission.submitted_by).to eq(user)
+        end
+
+        it 'records the submission completion time' do
+          submission_time = Time.zone.local(2018, 1, 10, 12, 13, 14)
+
+          travel_to(submission_time) do
+            post "/v1/submissions/#{submission.id}/complete", headers: { 'X-Auth-Id' => user.auth_id }
 
-        expect(submission.submitted_by).to eq(user)
+            submission.reload
+
+            expect(submission.submitted_at).to eq(submission_time)
+          end
+        end
       end
 
-      it 'records the submission completion time' do
-        submission_time = Time.zone.local(2018, 1, 10, 12, 13, 14)
+      context 'when there are completed submissions against the task' do
+        let(:task) { FactoryBot.create(:task, status: :completed) }
+        let!(:old_submission1) do
+          FactoryBot.create(
+            :submission_with_validated_entries,
+            aasm_state: 'completed',
+            task: task
+          )
+        end
+        let!(:old_submission2) do
```

Comment:
As above, we calcify (potentially for the lifetime of the project) support for multiple completed submissions by having it a requirement for this spec to pass.

Reply:
As above. Either we write code assuming good state, or write code defensively.